**Nombre:** Leonel Alessandro González Martínez
**Matrícula:** 81724

# Práctica 1.1 — Probabilidad y Estadística
## Análisis de la evolución poblacional mundial

En esta práctica se trabaja con datos reales de población obtenidos de **Our World in Data (OWID)**, cuya fuente original es **United Nations, World Population Prospects (2024)**.

El propósito es familiarizarse con un conjunto de datos real y comprobar que el entorno de trabajo (Python, Pandas, NumPy, Jupyter) funciona correctamente.

## Objetivo

Utilizar un conjunto de datos real para comprobar el funcionamiento del entorno de análisis de datos y aplicar medidas estadísticas descriptivas básicas utilizando Pandas y NumPy.

## Actividad 1 — Identificación de la fuente

**Propósito:** reconocer la procedencia del conjunto de datos antes de utilizarlo y documentar correctamente su fuente.

**Cita de la fuente:**

> UN, World Population Prospects (2024) – processed by Our World in Data. "Annual change in population – UN WPP" [dataset]. United Nations, "World Population Prospects"; United Nations, "World Population Prospects - Interim Update" [original data].

**Preguntas**

1. **¿Quién produjo originalmente los datos?** Naciones Unidas (United Nations), a través de World Population Prospects (2024).
2. **¿Qué organización los procesa para su utilización en Our World in Data?** Our World in Data (OWID) procesa y estandariza los datos originales de la ONU.
3. **¿Qué representa la variable que analizarás?** "Annual change in population" es el cambio *neto* anual de población: la diferencia entre la población al 1 de julio de dos años consecutivos, considerando nacimientos, muertes y migración en conjunto.

## Actividad 2 — Carga y exploración de datos

**Propósito:** comprobar que el entorno de Python, Pandas y Jupyter funciona correctamente y realizar una primera exploración del dataset.

### 2.1 Importar las bibliotecas

In [1]:
import pandas as pd
import numpy as np

### 2.2 Cargar el archivo CSV

In [2]:
dataframe = pd.read_csv("../data/raw/annual-population-growth/annual-population-growth.csv")

### 2.3 Explorar el DataFrame

In [3]:
dataframe.head(5)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
0,Afghanistan,AFG,1951,103163.0,NaN
1,Afghanistan,AFG,1952,108441.0,NaN
2,Afghanistan,AFG,1953,108919.0,NaN
3,Afghanistan,AFG,1954,111251.0,NaN
4,Afghanistan,AFG,1955,119027.0,NaN


In [4]:
dataframe.shape



(38400, 5)

In [5]:
dataframe.columns.tolist()

['Entity',
 'Code',
 'Year',
 'Annual change in population',
 'Annual population change (Projected)']

In [6]:
dataframe.dtypes


Entity                                      str
Code                                        str
Year                                      int64
Annual change in population             float64
Annual population change (Projected)    float64
dtype: object

In [7]:
df = dataframe

In [8]:
df.describe()

,Year,Annual change in population,Annual population change (Projected)
count,38400.000000,1.868800e+04,1.971200e+04
mean,2025.500000,2.063019e+06,8.375481e+05
std,43.300872,9.440859e+06,5.816745e+06
min,1951.000000,-3.315935e+06,-2.291035e+07
25%,1988.000000,1.781000e+03,-9.742750e+03
50%,2025.500000,4.689800e+04,9.350000e+01
75%,2063.000000,3.021128e+05,1.411160e+05
max,2100.000000,9.337137e+07,7.206736e+07


In [9]:
df.isnull().sum()

Entity                                      0
Code                                     1200
Year                                        0
Annual change in population             19712
Annual population change (Projected)    18688
dtype: int64

In [10]:
df["Entity"].nunique()

256

**Preguntas**

1. **¿Cuántas filas y columnas tiene el dataset?** 38,400 filas y 5 columnas.
2. **¿Qué tipo de datos contiene la variable `Year`?** `int64` (números enteros).
3. **¿Existen valores faltantes?** Sí: `Code` tiene 1,200 valores nulos, `Annual change in population` tiene 19,712 y `Annual population change (Projected)` tiene 18,688.
4. **¿Cuántas entidades diferentes contiene el dataset?** 256 (incluye países y también entidades agregadas como continentes o "World").

## Actividad 3 — Estadística descriptiva

**Propósito:** aplicar medidas estadísticas básicas a una variable poblacional.

Se selecciona el año **2020** (periodo histórico) y se filtran los datos correspondientes. Sobre la variable **"Annual change in population"** se calculan: media, mediana, varianza, desviación estándar, mínimo y máximo.

In [11]:
anio = 2020
df_anio = df[df["Year"] == anio]
df_anio.head(3)

,Entity,Code,Year,Annual change in population,Annual population change (Projected)
69,Afghanistan,AFG,2020,1212852.0,NaN
219,Africa (UN),UN_AFR,2020,32815675.0,NaN
369,Albania,ALB,2020,-13061.0,NaN


In [12]:
variable = "Annual change in population"

media = df_anio[variable].mean()
mediana = df_anio[variable].median()
varianza = df_anio[variable].var()
desviacion_estandar = df_anio[variable].std()
minimo = df_anio[variable].min()
maximo = df_anio[variable].max()

print("Media:", media)
print("Mediana:", mediana)
print("Varianza:", varianza)
print("Desviación estándar:", desviacion_estandar)
print("Mínimo:", minimo)
print("Máximo:", maximo)

Media: 2124908.84765625
Mediana: 34293.5
Varianza: 90687335031103.64
Desviación estándar: 9522989.815761836
Mínimo: -494016.0
Máximo: 75707590.0


In [13]:
entidad_min = df_anio.loc[df_anio[variable].idxmin(), "Entity"]
entidad_max = df_anio.loc[df_anio[variable].idxmax(), "Entity"]

print("Entidad con el valor mínimo:", entidad_min)
print("Entidad con el valor máximo:", entidad_max)

Entidad con el valor mínimo: Venezuela
Entidad con el valor máximo: World


**Preguntas**

1. **¿Cuál es el valor de la media?** ≈ 2,124,908.85 personas.
2. **¿Cuál es el valor de la mediana?** 34,293.5 personas.
3. **¿Cuál es el valor de la desviación estándar?** ≈ 9,522,989.82 personas.
4. **¿Qué entidad presenta el valor mínimo y cuál el máximo?** El valor mínimo lo presenta **Venezuela** (−494,016) y el máximo lo presenta **"World"** (75,707,590). Nota: "World" es una entidad agregada (no un país individual), por lo que representa la suma global, no el país con mayor crecimiento.

## Actividad 4 — Comprobación del entorno

**Propósito:** comprobar que el notebook está utilizando correctamente las herramientas configuradas para la práctica.

In [14]:
import sys

print("Python:", sys.version)
print("Pandas:", pd.__version__)
print("NumPy:", np.__version__)

Python: 3.12.10 (tags/v3.12.10:0cc8128, Apr  8 2025, 12:21:36) [MSC v.1943 64 bit (AMD64)]
Pandas: 3.0.5
NumPy: 2.5.2


Se verifica que las versiones mostradas corresponden al entorno virtual (`.venv`) configurado para la práctica. Finalmente, se debe ejecutar **Restart Kernel and Run All Cells** para comprobar que todo el notebook se reproduce correctamente de principio a fin.

## Análisis adicional — País con "A" (2006–2026)

Se filtran los datos de **Argentina** entre los años 2006 y 2026, y se calculan media, mediana, varianza y desviación estándar de "Annual change in population".

In [15]:
pais = "Argentina"
df_pais_fechas = df[(df["Entity"] == pais) & (df["Year"] >= 2006) & (df["Year"] <= 2026)]
df_pais_fechas[["Entity", "Year", "Annual change in population"]]

,Entity,Year,Annual change in population
1555,Argentina,2006,405327.0
1556,Argentina,2007,394654.0
1557,Argentina,2008,407384.0
1558,Argentina,2009,430681.0
1559,Argentina,2010,433862.0
1560,Argentina,2011,441961.0
1561,Argentina,2012,431063.0
1562,Argentina,2013,420735.0
1563,Argentina,2014,441618.0
1564,Argentina,2015,452940.0


In [16]:
variable_pais = df_pais_fechas["Annual change in population"]

media_pais = variable_pais.mean()
mediana_pais = variable_pais.median()
varianza_pais = variable_pais.var()
desviacion_estandar_pais = variable_pais.std()

print(f"Media ({pais}, 2006-2026):", media_pais)
print(f"Mediana ({pais}, 2006-2026):", mediana_pais)
print(f"Varianza ({pais}, 2006-2026):", varianza_pais)
print(f"Desviación estándar ({pais}, 2006-2026):", desviacion_estandar_pais)

Media (Argentina, 2006-2026): 351200.8888888889
Mediana (Argentina, 2006-2026): 406355.5
Varianza (Argentina, 2006-2026): 14819162677.281046
Desviación estándar (Argentina, 2006-2026): 121733.98324741142


## Resultados

- **Fuente:** United Nations, World Population Prospects (2024), procesado por Our World in Data.
- **DataFrame cargado correctamente:** 38,400 filas × 5 columnas, con 256 entidades distintas.
- **Estructura:** columnas `Entity`, `Code`, `Year`, `Annual change in population` y `Annual population change (Projected)`; `Year` es `int64` y las columnas numéricas de cambio poblacional son `float64`, con valores faltantes en `Code` y en ambas variables de cambio poblacional.
- **Estadísticos (año 2020, "Annual change in population"):** media ≈ 2,124,908.85, mediana = 34,293.5, desviación estándar ≈ 9,522,989.82, mínimo = −494,016 (Venezuela), máximo = 75,707,590 ("World").
- **Entorno verificado:** versiones de Python, Pandas y NumPy comprobadas en la Actividad 4, y el notebook se reproduce correctamente al ejecutarse desde el inicio.